kin-core/src/main/java/kin/core/AccountInfoRetriever.java

```diff
@@ -62,6 +62,37 @@ Balance getBalance(@NonNull String accountId) throws OperationFailedException {
         return balance;
     }
 
+    /**
+     * Check if the account has been "burned".
+     * @param accountId the account ID to check if it is "burned"
+     * @return true if the account is "burned", false otherwise.
+     * @throws AccountNotFoundException if account not created yet
+     * @throws AccountNotActivatedException if account has no Kin trust
+     * @throws OperationFailedException any other error
+     */
+    boolean isAccountBurned(@NonNull String accountId) throws OperationFailedException {
+        Utils.checkNotNull(accountId, "account");
+        boolean isBurned;
+        try {
+            AccountResponse accountResponse = server.accounts().account(KeyPair.fromAccountId(accountId));
+            if (accountResponse == null) {
+                throw new OperationFailedException("can't retrieve data for account " + accountId);
+            }
+            AccountResponse.Signer signer = accountResponse.getSigners()[0];
```

Comment:
It is possible from the blockchain perspective but for us there is no such use case.

Reply:
I think we should check for all signers to avoid some edge cases/hacks.
@Ronserruya do we check for all signers in migration service side?

Reply:
My answer was after i consult also with Ron.
But anyway lets wait for @Ronserruya  answer to your question.

Reply:
@yosriz No, we only check the master signer, since we expect that the client will only have one signer.

I think the client can stay like this. However I do think that I need to change the migration server, since the migration server will be accessible not only be clients

Reply:
yes, I tend to agree, migration server must do it, client not as ms will check it. @drEfrati @Ronserruya I know it's an edge case, but we should have a proper error from the migration server, so the client can report this error in BI,log etc.

Reply:
So I am merging this and @Ronserruya if you implement it then just tell me what is the new error and i will deal with it the client